x/accesscontrol/keeper/keeper.go

```diff
@@ -0,0 +1,73 @@
+package keeper
+
+import (
+	"fmt"
+
+	"github.com/tendermint/tendermint/libs/log"
+
+	"github.com/cosmos/cosmos-sdk/codec"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+
+	"github.com/cosmos/cosmos-sdk/x/accesscontrol/types"
+
+	paramtypes "github.com/cosmos/cosmos-sdk/x/params/types"
+)
+
+type (
+	Keeper struct {
+		cdc        codec.BinaryCodec
+		storeKey   sdk.StoreKey
+		paramSpace paramtypes.Subspace
+	}
+)
+
+func NewKeeper(
+	cdc codec.Codec,
+	storeKey sdk.StoreKey,
+	paramSpace paramtypes.Subspace,
+) Keeper {
+	if !paramSpace.HasKeyTable() {
+		paramSpace = paramSpace.WithKeyTable(types.ParamKeyTable())
+	}
+
+	return Keeper{
+		cdc:        cdc,
+		storeKey:   storeKey,
+		paramSpace: paramSpace,
+	}
+}
+
+func (k Keeper) Logger(ctx sdk.Context) log.Logger {
+	return ctx.Logger().With("module", fmt.Sprintf("x/%s", types.ModuleName))
+}
+
+func (k Keeper) GetResourceDepedencyMapping(ctx sdk.Context, moduleName string, messageType string) types.MessageDependencyMapping {
```

Comment:
lets take in a single key instead of module name and message type, and then we can just call the helper on the Msg to give us the accesscontrol key.